.github/workflows/check_pr_for_broken_links.yml

```diff
@@ -0,0 +1,40 @@
+name: CheckPRLinks
+on:
+  pull_request:
+    branches: [main]
+    types: [opened, synchronize]
+  workflow_dispatch:
+permissions:
+  contents: read
+  id-token: write
```

Comment:
Do we need these permissions for this specific workflow? I think we only need it when we're using the `configure-aws-actions` script right?